r/R/worker_helpers.R

```diff
@@ -97,3 +97,26 @@ getClusters <- function(type, resolution, data) {
   }
   return(data)
 }
+
+handle_pagination <- function(gene_results, offset, limit, order_by, order_decreasing, filter = NULL) {
+  if (!is.null(filter)) {
+    gene_results <- gene_results[grepl(filter, gene_results$gene_name, ignore.case = TRUE), ]
+  }
+
+  if(nrow(gene_results)>0){
+    gene_results$full_count <- nrow(gene_results)
+  }else{
+    gene_results$full_count <- list()
```

Comment:
Shouldn't you just return 0 instead of an empty list? Doesn't `nrow` return a number / integer?